journal.py

```diff
@@ -148,77 +177,50 @@ def markd(input):
 
 
-def write_entry(request):
-    """write a single entry to the database"""
-    title = request.params.get('title', None)
-    text = request.params.get('text', None)
-    created = datetime.datetime.utcnow()
-    request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
+# def write_entry(request):
+#     """write a single entry to the database"""
+#     title = request.params.get('title', None)
+#     text = request.params.get('text', None)
+#     created = datetime.datetime.utcnow()
+#     request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
 
 
 @view_config(route_name='home', renderer='templates/list.jinja2')
 def read_entries(request):
     """return a list of all entries as dicts"""
-    cursor = request.db.cursor()
-    cursor.execute(DB_ENTRIES_LIST)
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
-    for entry in entries:
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
+    entries = Entry.all()
     return {'entries': entries}
 
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_entry(request):
-    """return a single entry"""
-    cursor = request.db.cursor()
-    cursor.execute(SELECT_ID, (request.matchdict['id'], ))
-    keys = ('id', 'title', 'text', 'created')
-    row = cursor.fetchone()
-    entry = dict(zip(keys, row))
-    entry['text'] = markdown.markdown(
-        entry['text'], extensions=['codehilite', 'fenced_code'])
+    """return a single entry by its id"""
+    entry = Entry.by_id(request.matchdict['id'])
     return {'entry': entry}
 
 
-def update_entry(request):
-    id = request.params.get('id')
-    title = request.params.get('title')
-    text = request.params.get('text')
-    request.db.cursor().execute(UPDATE_ENTRY, [title, text, id])
+# def update_entry(request):
+#     id = request.params.get('id')
+#     title = request.params.get('title')
+#     text = request.params.get('text')
+#     request.db.cursor().execute(UPDATE_ENTRY, [title, text, id])
 
 
 @view_config(route_name='edit', renderer='json')
-def editview_entry(request):
+def edit_entry(request):
+    """gets posts by id and allows for editing"""
     if request.authenticated_userid:
+        entry = Entry.by_id(request.params.get('id', None))
         if request.method == 'GET':
-            cursor = request.db.cursor()
-            cursor.execute(SELECT_ID, (request.params.get('id', None), ))
-            keys = ('id', 'title', 'text', 'created')
-            row = cursor.fetchone()
-            entry = dict(zip(keys, row))
-            entry['created'] = entry['created'].strftime('%b %d, %Y')
-
-            return entry
+            return entry.json_edit()
 
         elif request.method == 'POST':
             try:
-                update_entry(request)
+                entry.editing(request)
             except psycopg2.Error:
```

Comment:
Watch out.  This exception handler will no longer work, since the errors will be raised by SQLAlchemy rather than psycopg2